qiita_db/portal.py

```diff
@@ -32,32 +33,156 @@ def get_studies_by_portal(portal):
         return {x[0] for x in
                 conn_handler.execute_fetchall(sql, [portal_id])}
 
-    def add_study_to_portal(study, portal):
-        """Adds study to given portal
+    def add_studies_to_portal(portal, studies):
+        """Adds studies to given portal
 
         Parameters
         ----------
+        studies : list of int
+            Study ids to attach to portal
         portal : str
             Portal to associate with.
         """
         portal_id = convert_to_id(portal, 'portal_type', 'portal')
+        conn_handler = SQLConnectionHandler()
+        sql = """SELECT study_id from qiita.study_portal
```

Comment:
I don't fully understand this query. If I'm reading correctly, it is checking for the studies that the portal id is different from the current portal that you want to add and the study id is passed in the current list. I think the thought process here is pretty complex... Shouldn't you just check if adding duplicates? For checking duplicates its easier to do:
```python
sql = "SELECT study_id FROM qiita.study_portal WHERE portal_type_id=%s AND study_id IN %s"
duplicates = [x[0] for x in conn_handler.execute_fetchall(sql, [portal_id, tuple(studies)])]

if duplicates:
   warnings.warn("...")
```

And I'd say that we want to do an error here rather than a warning. These operation will be executed by sysadmins so it's better to not allow to execute something if there is an error.

What is your rationale to do a warning rather than an error?

@biocore/qiita-dev can somebody else weight here if this should be an error or a warning?